### PR TITLE
fixes related to dynamic interfaces

### PR DESCRIPTION
**Pull request by olivier-matz-6wind**
Hi,

Please find two fixes related to dynamic interfaces.

Olivier

### CODE REVIEW

File: keepalived/vrrp/vrrp_parser.c [hunk 1]
```diff
@@ -461,6 +461,7 @@ vrrp_int_handler(vector_t *strvec)
 		return;
 	}
 
+	strcpy(vrrp->ifname, name);
```
Comment (acassen) [inline keepalived/vrrp/vrrp_parser.c]: strncpy() would be safer

Comment (olivier-matz-6wind) [reply keepalived/vrrp/vrrp_parser.c]: There is a check just above: if (strlen(name) >= IFNAMSIZ), I did like in vrrp_vmac_handler().

Please let me know if you still want me to update the pull request with strncpy(dst, src, sizeof(dst)), or with snprintf(dst, sizeof(dst), "%s", src)

Thanks for the fast review!

**Comment by acassen** [comment]
@olivier-matz-6wind : better w8 until Quentin review (even if strncpy got my flavor ;))

**Comment by pqarmitage** [comment]
@olivier-matz-6wind I've slightly modified your patches in line with my comments above, and have merged the updated patches.

Many thanks for doing all the hard work by identifying the bugs and providing the patches; all I have done is some slight streamlining of your patches to be more aligned with how keepalived works.

If you see any issues with the patches I have committed, please either report an issue or submit some further patches.

Many thanks for your help.

**Comment by olivier-matz-6wind** [comment]
Many thanks @pqarmitage and sorry for late feedback, I was off the last few days. The two reworked fixes look indeed better. I'll validate them on my platform asap and let you know.

**Comment by olivier-matz-6wind** [comment]
I finally found some time to validate the fixes, I confirm they properly solve the initial issues I've seen.

**Pull request closed without merging**